bigframes/dataframe.py

```diff
@@ -1066,6 +1066,14 @@ def head(self, n: int = 5) -> DataFrame:
     def tail(self, n: int = 5) -> DataFrame:
         return typing.cast(DataFrame, self.iloc[-n:])
 
+    def peek(self, n: int = 5) -> pandas.DataFrame:
+        maybe_result = self._block.try_peek(n)
+        if maybe_result is None:
+            raise NotImplementedError(
+                "Cannot peek efficiently when data has aggregates, joins or window functions applied."
```

Comment:
It's not always going to be obvious when the data has aggregates, windows, or especially not joins (due to implicit joins on index).

Could we instead call `_cached()` in this case and then peek?

Reply:
Added a force=True param to peek. This will cause the dataframe to execute and cache the block if it is not peekable. Are you sure this is the best default behavior? One of my goals with peek was for users to avoid fully computing the dataframe.

Reply:
In general, I'd say the pandas API skews towards the most usable default, not the most efficient one. That said, our main motivation here is to address the feedback of how expensive `head()` can be, so I see the argument for having `force=False` the default.